From 532a3c77ce735fae7f7155cae5fbda768120fc1e Mon Sep 17 00:00:00 2001
From: Joshua Covington <joshuacov@gmail.com>
Date: Wed, 1 Aug 2026 20:54:52 +0000
Subject: [PATCH] Post-Merge Fixup - up to
 825dba6c63eeb40a62699d1f8a4aa3f02b0eaf49

The man page for 'git format-patch' states that it is not suitable for dealing
with merge commits, which are omitted by default. There is no way to avoid
conflicts when generating a patch from a non-linear history with overlapping
changes (e.g., when multiple branches include the same commit).

Due to frequent merges to and from the mainline kernel, exporting patches from
rpi-6.18.y using 'git format-patch' does not capture the full state of the
tree. This diff represents the manual fixups required where those merge
commits occurred. Applying this diff alongside the previously exported patches
yields a complete, accurate copy of the official rpi-6.18.y branch.

Note: This last patch must be updated after every export from the rpi-6.18.y
branch using 'git format-patch'.

Signed-off-by: Joshua Covington <joshuacov@gmail.com>
---

 arch/arm64/kernel/cpu_errata.c                |    2 +-
 drivers/gpu/drm/v3d/v3d_submit.c              |    2 +-
 drivers/mmc/core/card.h                       |   10 +-
 drivers/nvme/host/pci.c                       |    4 +-
 4 files changed, 9 insertions(+), 9 deletions(-)

--- a/arch/arm64/kernel/cpu_errata.c
+++ b/arch/arm64/kernel/cpu_errata.c
@@ -699,7 +699,7 @@ const struct arm64_cpu_capabilities arm6
 #endif
 #ifdef CONFIG_ARM64_WORKAROUND_REPEAT_TLBI
 	{
-		.desc = "Qualcomm erratum 1009, or ARM erratum 1286807, 2441009",
+		.desc = "Broken broadcast TLBI completion",
 		.capability = ARM64_WORKAROUND_REPEAT_TLBI,
 		.type = ARM64_CPUCAP_LOCAL_CPU_ERRATUM,
 		.matches = cpucap_multi_entry_cap_matches,
--- a/drivers/gpu/drm/v3d/v3d_submit.c
+++ b/drivers/gpu/drm/v3d/v3d_submit.c
@@ -1352,7 +1352,7 @@ v3d_submit_cpu_ioctl(struct drm_device *
 	trace_v3d_submit_cpu_ioctl(&v3d->drm, cpu_job->job_type);
 
 	ret = v3d_job_init(v3d, file_priv, &cpu_job->base,
-			   v3d_job_free, 0, &se, V3D_CPU);
+			   v3d_cpu_job_free, 0, &se, V3D_CPU);
 	if (ret) {
 		v3d_job_deallocate((void *)&cpu_job);
 		goto fail;
--- a/drivers/mmc/core/card.h
+++ b/drivers/mmc/core/card.h
@@ -309,6 +309,11 @@ static inline int mmc_card_no_uhs_ddr50_
 	return c->quirks & MMC_QUIRK_NO_UHS_DDR50_TUNING;
 }
 
+static inline int mmc_card_working_sd_cq(const struct mmc_card *c)
+{
+	return c->quirks & MMC_QUIRK_WORKING_SD_CQ;
+}
+
 static inline int mmc_card_broken_mdt(const struct mmc_card *c)
 {
 	return c->quirks & MMC_QUIRK_BROKEN_MDT;
@@ -319,9 +324,4 @@ static inline int mmc_card_fixed_secure_
 	return c->quirks & MMC_QUIRK_FIXED_SECURE_ERASE_TRIM_TIME;
 }
 
-static inline int mmc_card_working_sd_cq(const struct mmc_card *c)
-{
-	return c->quirks & MMC_QUIRK_WORKING_SD_CQ;
-}
-
 #endif
--- a/drivers/nvme/host/pci.c
+++ b/drivers/nvme/host/pci.c
@@ -768,8 +768,8 @@ static void nvme_unmap_data(struct reque
 
 	if (!blk_rq_dma_unmap(req, dma_dev, &iod->dma_state, iod->total_len)) {
 		if (nvme_pci_cmd_use_sgl(&iod->cmd))
-			nvme_free_sgls(req, iod->descriptors[0],
-				       &iod->cmd.common.dptr.sgl);
+			nvme_free_sgls(req, &iod->cmd.common.dptr.sgl,
+				       iod->descriptors[0]);
 		else
 			nvme_free_prps(req);
 	}
